docs/developer-docs/setup/pulling-canister-dependencies.md

```diff
@@ -0,0 +1,441 @@
+init_guide => A natural number, e.g. 20.
+candid:args => (nat)
+```
+
+To set an init argument with an argument using the `--argument` flag, the following commands can be used:
+
+```
+dfx deps init yofga-2qaaa-aaaaa-aabsq-cai --argument 10
+dfx deps init deps_c --argument 20
+```
+
+The resulting generated file `init.json` will have the following content:
+
+```json
+{
+  "canisters": {
+    "yofga-2qaaa-aaaaa-aabsq-cai": {
+      "arg_str": "10",
+      "arg_raw": "4449444c00017d0a"
+    },
+    "yhgn4-myaaa-aaaaa-aabta-cai": {
+      "arg_str": null,
+      "arg_raw": null
+    },
+    "yahli-baaaa-aaaaa-aabtq-cai": {
+      "arg_str": "20",
+      "arg_raw": "4449444c00017d14"
+    }
+  }
+}
+```
+
+### Step 4: Deploy the pulled dependencies on a local replica using the `dfx deps deploy` command.
+
+Running the `dfx deps deploy` command will:
+
+1. First, create the dependencies on the local replica with the same mainnet canister ID.
+2. Then, it will install the downloaded wasm with the init arguments in the `init.json` file.
+
+You can also specify the name or principal to deploy one particular dependency.
+
+Using the example above, we can run the following command to deploy all dependencies:
+
+```
+dfx deps deploy
+Creating canister: yofga-2qaaa-aaaaa-aabsq-cai
+Installing canister: yofga-2qaaa-aaaaa-aabsq-cai
+Creating canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+Installing canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+Creating canister: yahli-baaaa-aaaaa-aabtq-cai (dep_c)
+Installing canister: yahli-baaaa-aaaaa-aabtq-cai (dep_c)
+```
+
+To deploy one particular dependency, the following command can be used: 
+
+```
+dfx deps deploy yofga-2qaaa-aaaaa-aabsq-cai
+Installing canister: yofga-2qaaa-aaaaa-aabsq-cai
+```
+
+```
+dfx deps deploy dep_b
+Installing canister: yhgn4-myaaa-aaaaa-aabta-cai (dep_b)
+```
+
+:::info
+`dfx deps deploy` always creates the canister with the anonymous identity so that dependencies and application canisters will have different controllers. It also always installs the canister in "reinstall" mode so that the canister status will be discarded.
+:::
+
+## Interactive example
+
+Now that we've explored the concepts and overview of using the `dfx deps` workflow, let's take a look at using an interactive example to demonstrate the functionality. 
+
+This example project will demonstrate an application canister pulling its dependency from the IC mainnet and integrating with it locally. 
+
+In this example, the `app` canister defines a method called `double_service` which makes an inter-canister call to the `service` canister.
+
+- #### Step 1: First, assure that you have installed the [IC SDK](https://github.com/dfinity/sdk) and [git](https://git-scm.com/downloads).
+
+- #### Step 2: Then, open a terminal window and use `git clone` to download the following repository:
+
+```
+git clone https://github.com/lwshang/pullable.git
+```
+
```

Comment:
Clone the service provider's repo is not a required step.

In this demo, a service consumer should `dfx new` to get new project and edit the `dfx.json` as in step 3 below.